From 3468310486d0da8d2a6c6bdcddc598b7b8ce5587 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Tue, 4 Feb 2020 14:18:01 +0100
Subject: [PATCH] Revert "wait for IPv6 RA if using none/static IPv6
 assignment"

This reverts commit c603419030136570b5944dc4620f62d07b9e82bb.

wait_for_ipv6_dad_link is only called from dhclient script,
so the original intent "wait for IPv6 RA if using none/static IPv6 assignment"
does not seem to be correct.

Anyway, this brings an issue on isolated networks, where you don't
have any routes outside. dhclient-script hangs on this check
and after it times out, dhclient is able to set the address normally.

(cherry picked from commit 76f6566fd777e0ee20416061ecb795e8daef33e1)

Resolves: #1795276
---
 modules.d/40network/net-lib.sh | 1 -
 1 file changed, 1 deletion(-)

diff --git a/modules.d/40network/net-lib.sh b/modules.d/40network/net-lib.sh
index dcb53804..0f3ca6a8 100755
--- a/modules.d/40network/net-lib.sh
+++ b/modules.d/40network/net-lib.sh
@@ -655,7 +655,6 @@ wait_for_ipv6_dad_link() {
     while [ $cnt -lt $timeout ]; do
         [ -n "$(ip -6 addr show dev "$1" scope link)" ] \
             && [ -z "$(ip -6 addr show dev "$1" scope link tentative)" ] \
-            && [ -n "$(ip -6 route list proto ra dev "$1" | grep ^default)" ] \
             && return 0
         [ -n "$(ip -6 addr show dev "$1" scope link dadfailed)" ] \
             && return 1

